primitives/trie/src/lib.rs

```diff
@@ -186,6 +186,9 @@ pub fn delta_trie_root<L: TrieConfiguration, I, A, B, DB, V>(
 	{
 		let mut trie = TrieDBMut::<L>::from_existing(&mut *db, &mut root)?;
 
+		let mut delta = delta.into_iter().collect::<Vec<_>>();
```

Comment:
I don't know if it is worth it to describe in the function doc that it internally sort input delta.
This would help track a possible micro optimisation.
As I say, not sure if worth it.